File: Easy/1370. Increasing Decreasing String/sortString.java
class Solution {
    public String sortString(String s) {
        int[] count = new int[26];
        for(int i = 0; i < s.length(); i++){
            count[s.charAt(i) - 'a']++;
        }
        StringBuilder sb = new StringBuilder();
        while(sb.length() < s.length()){
            for(int i = 0; i < 26; i++){
                if(count[i] > 0){
                    sb.append((char)('a' + i));
                    count[i]--;
                }
            }
            for(int i = 25; i >= 0; i--){
                if(count[i] > 0){
                    sb.append((char)('a' + i));
                    count[i]--;
                }
            }
        }
        return sb.toString();
    }
}

File: Easy/1370. Increasing Decreasing String/sortString.cpp
class Solution {
public:
    string sortString(string s) {
        /*
            time complexity:O(n)
            space complexity:O(n)
        */
        vector<int> count(26,0);
        int n = s.size();
        for(char c : s){
            count[c - 'a']++;
        }
        string ans = "";
        while(n > 0){
            //從'a'到'z'開始找找到不能找
            for(int i = 0; i < 26; i++){
                if(count[i] > 0){
                    ans += ('a' + i);
                    count[i]--;
                    n--;
                }
            }
            //從'z'到'a'開始找找到不能找
            for(int i = 25; i >= 0; i--){
                if(count[i] > 0){
                    ans += ('a' + i);
                    count[i]--;
                    n--;
                }
            }
        }
        return ans;
    }
};